Polygon_mesh_processing/include/CGAL/Rigid_triangle_mesh_collision_detection.h

```diff
@@ -176,6 +176,25 @@ class Rigid_triangle_mesh_collision_detection
     : m_free_id(0)
   {}
 
+  /**
+    * Constructor moving an instance of Rigid_triangle_mesh_collision_detection to a new memory
+    * location with minimal memory copy.
+    * @param other The instance to be moved
+    */
+  Rigid_triangle_mesh_collision_detection(Rigid_triangle_mesh_collision_detection&& other)
+    :
+      m_own_aabb_trees {std::move(other.m_own_aabb_trees)}
+    ,m_aabb_trees {std::move(other.m_aabb_trees)}
+    ,m_is_closed {std::move(other.m_is_closed)}
+    ,m_points_per_cc {std::move(other.m_points_per_cc)}
+    ,m_traversal_traits {std::move(other.m_traversal_traits)}
+    ,m_free_id {std::move(other.m_free_id)}
+    ,m_id_pool {std::move(other.m_id_pool)}
+  {
+    for(std::size_t i = 0; i< other.m_own_aabb_trees.size(); ++i)
+      other.m_own_aabb_trees[i]= false;
+  }
+
```

Comment:
In `AABB_tree` we simply do (i.e. use move assignement):
```
  template<typename Tr>
  AABB_tree<Tr>::AABB_tree(Self&& tree) noexcept
  {
    *this = std::move(tree);
  }
  ```
  I guess we could do the same to avoid duplication. Do you agree @lrineau ?

Reply:
Yes, I agree.